tapdb/sqlc/queries/assets.sql

```diff
@@ -676,6 +685,16 @@ WHERE (
 DELETE FROM managed_utxos
 WHERE outpoint = $1;
 
+-- name: LeaseManagedUTXO :exec
+UPDATE managed_utxos
+SET lease_owner = @lease_owner, lease_expiry = @lease_expiry
```

Comment:
See above about making this an insert. That way, we can have the unique key be the outpoint itself. Then if two callers try to lease the same UTXO, they get an error. Otherwise, with the current API the same coin can be leased twice.

Reply:
See comment above. I think it makes things unnecessarily hard.
And a coin cannot be leased twice because of the wallet-level coin lock:
 1. Caller A comes in, does: `WalletLock( ListEligibleCoins => returns coins X, Y, Z => SelectForAmount => select coin X => LeaseCoin(X))`
 2. Caller B comes in while call A is still ongoing. Once it acquires the lock, it does: `WalletLock( ListEligibleCoins => returns coins Y, Z => SelectForAmount => select coin Y => LeaseCoin(Y))`

Reply:
Ah yeh that's indeed the case with the full API path as exists today. 

I think I was just viewing it more from the PoV of the lowest level data structure here: the table itself. If we used the `outpoint` (or `outpoint || assetID` or w/e) as a primary key, then we'd be able to entirely rely on the DB to preserve the invariant that something can only be leased once: another insert would fail. As is, it would be possible for to callers (strictly on the db level just to zoom in) to each lease the same UTXO, with the second replacing/mutating the `lease_owner` field. 

As you mention above it still works as is, but if we wanted to try and normalize things further, I think we'd put the leases in a distinct table. Though up until now, things were normalized mostly on a best effort basis, so we don't necessarily have a strict policy here. Either way, just wanted to point this out as we recently-ish learned that the `sqlite` migration support isn't as fully featured as `postgres` so it'll be more difficult to change later.